cristin-import/src/main/java/no/unit/nva/cristin/mapper/Nsd.java

```diff
@@ -22,26 +24,55 @@ public Nsd(int nsdCode, int year, ChannelRegistryMapper channelRegistryMapper) {
         this.channelRegistryMapper = channelRegistryMapper;
     }
 
-    public URI createJournalOrSeriesUri() {
-        return lookupNsdJournalOrSeriesProxyUri()
-                   .orElseGet(() -> lookupNsdPublisherProxyUri()
-                                        .orElseThrow(() -> new ChannelRegistryException(nsdCode)));
-    }
-
     public URI getPublisherUri() {
         return lookupNsdPublisherProxyUri()
                    .orElseGet(() -> lookupNsdJournalOrSeriesProxyUri()
                                         .orElseThrow(() -> new ChannelRegistryException(nsdCode)));
     }
 
+    public URI createJournal() {
+        return lookUpNsdJournal().orElseThrow(() -> new ChannelRegistryException(nsdCode));
+    }
+
+    public URI createSeries() {
+        return lookUpNsdSeries().orElseThrow(() -> new ChannelRegistryException(nsdCode));
+    }
+
+    private Optional<URI> lookUpNsdSeries() {
+        return channelRegistryMapper.convertNsdJournalCodeToPid(nsdCode)
+            .map(this::toSeriesUri);
+    }
+
+    private URI toSeriesUri(ChannelRegistryEntry channelRegistryEntry) {
+        if (ChannelType.SERIES.equals(channelRegistryEntry.type())) {
+            return getNsdProxyUri(channelRegistryEntry.getEntryPath(), channelRegistryEntry.id());
+        } else {
+            throw new WrongChannelTypeException(channelRegistryEntry);
+        }
```

Comment:
Throwing exception when type from channelRegistry is journal, but NVA publication can only have series.

Reply:
Blir spennende å lage WrongChannelTypeException retteliste ved neste kjøring.